subsys/nrf_rpc/nrf_rpc_uart.c

```diff
@@ -176,39 +184,49 @@ static bool crc_compare(uint16_t rx_crc, uint16_t calc_crc)
 	return rx_crc == calc_crc;
 }
 
-static void hdlc_decode_byte(struct nrf_rpc_uart *uart_tr, uint8_t byte)
+static void hdlc_decode_byte(enum hdlc_state *state, uint8_t *out, size_t *len, size_t capacity,
```

Comment:
please stick with struct `nrf_rpc_uart *uart_tr`.
Returning through pointers is bug prone and you (as in this case) quickly go beyond 4 function args

Reply:
Yeah, but I wanted to reuse this function for 2 decoding states (for decoding ACKs in ISR and long packets in work queue). I thought about abstracting `struct decoding_state` and pass just one member instead of all these pointers, but this would result in a small RAM increase. But I think the increase is acceptable in this case for the sake of cleaner code.

Reply:
Added `struct hdlc_decode_ctx` to reduce the number of input arguments.